ui/src/AutomationService/Group.purs

```diff
@@ -53,12 +55,16 @@ decodeGroup groupJson = do
       [] -> Right []
       grs' -> sequence grs'
 
-    decodeMember :: Json -> Either JsonDecodeError GroupDevice
-    decodeMember memberJson = do
+    decodeMember :: Devices -> Json -> Either JsonDecodeError GroupDevice
+    decodeMember devices' memberJson = do
       obj <- decodeJson memberJson
       endpoint <- obj .: "endpoint"
       id <- obj .: "ieee_address"
-      pure { id, endpoint }
+      case M.lookup id devices' of
+        Just device ->
+          pure { id, endpoint, device }
```

Comment:
I'm a little nervous about Groups having their own copy of Devices, but I think the decode flow in Main (Groups get rebuilt whenever Devices are decoded successfully) should ensure this doesn't happen. But if it's not a pain to do so, it may make more sense to look Devices up when we need them when we're working with Groups.